crates/oxc_semantic/src/binder.rs

```diff
@@ -439,6 +439,39 @@ impl<'a> Binder<'a> for TSModuleDeclaration<'a> {
 
 impl<'a> Binder<'a> for TSTypeParameter<'a> {
     fn bind(&self, builder: &mut SemanticBuilder) {
+        if matches!(
```

Comment:
block based on:

https://github.com/typescript-eslint/typescript-eslint/pull/2497/files#diff-0cfd7f7e671424ab7fca326415a261843e64acdd13ecb12fc5133bd98b642948

Reply:
It looks like the `typescript-eslint` implementation has more checks, but our ScopeTree and SymbolTable are based on TypeScript, so we can simplify that like https://github.com/microsoft/TypeScript/blob/15392346d05045742e653eab5c87538ff2a3c863/src/compiler/binder.ts#L3764-L3767 does